libs/src/sanityChecks/rolloverNodes.js

```diff
@@ -1,69 +1,88 @@
   const user = libs.userStateManager.getCurrentUser()
 
   if (!user || !user.is_creator) return
 
   const primary = CreatorNode.getPrimary(user.creator_node_endpoint)
-  const healthy = await checkPrimaryHealthy(libs, primary, MAX_TRIES)
-  if (healthy) return
-
   const secondaries = CreatorNode.getSecondaries(user.creator_node_endpoint)
 
   try {
-    // Get a new primary
-    const newPrimary = await getNewPrimary(libs, secondaries)
-    const index = secondaries.indexOf(newPrimary)
-    // Get new secondaries and backfill up to 2
+    let newPrimary = primary
     let newSecondaries = [...secondaries]
-    newSecondaries.splice(index, 1)
-    const autoselect = await libs.ServiceProvider.autoSelectCreatorNodes(
+
+    // Select new primary from secondaries if necessary, and remove that node from secondaries list.
+    if (!await nodeIsHealthy(primary, MAX_TRIES)) {
+      const upToDateSecondary = await getUpToDateSecondary(libs, secondaries)
+      if (!upToDateSecondary) {
+        throw new Error('Cannot select new primary - no valid secondaries found for user.')
+      }
+      newPrimary = upToDateSecondary
+      newSecondaries.splice(secondaries.indexOf(upToDateSecondary), 1)
+    }
+
+    // Filter out unhealthy secondaries.
+    // TODO - see if possible to replace with filter with await isHealthy check
+    const unhealthySecondaries = []
+    for (const secondary of newSecondaries) {
+      if (!await nodeIsHealthy(secondary, MAX_TRIES)) {
+        unhealthySecondaries.push(secondary)
+      }
+    }
+    newSecondaries = newSecondaries.filter(s => !unhealthySecondaries.includes(s))
+
+    if (newSecondaries.length >= 2 && _.isEqual(primary, newPrimary) && _.isEqual(secondaries, newSecondaries)) {
+      return
+    }
+
+    // Backfill secondaries as needed.
+    const { primary: autoPrimary, secondaries: autoSecondaries } = await libs.ServiceProvider.autoSelectCreatorNodes(
       2 - newSecondaries.length,
       creatorNodeWhitelist,
-      // Exclude ones we currently have
-      new Set([newPrimary, ...newSecondaries])
+      // Exclude ones we currently have.
+      new Set([newPrimary, ...newSecondaries]),
+      /** enforceSyncStatus */ false
     )
-    newSecondaries = newSecondaries.concat([autoselect.primary, ...autoselect.secondaries])
-
-    // Set the new endpoint and connect to it
+    newSecondaries = newSecondaries.concat([autoPrimary, ...autoSecondaries])
     const newEndpoints = [newPrimary, ...newSecondaries]
-    await libs.creatorNode.setEndpoint(newEndpoints[0])
 
-    // Update the user
+    // Set and connect to new primary endpoint.
+    await libs.creatorNode.setEndpoint(newPrimary)
+
+    // Update user's state with new nodes.
     const newMetadata = { ...user }
     newMetadata.creator_node_endpoint = newEndpoints.join(',')
     await libs.User.updateCreator(user.user_id, newMetadata)
   } catch (e) {
-    console.error(e)
+    console.error(`Rollover Nodes error: ${e}`)
```

Comment:
is this retried in the caller? because the outcome of this can vary with availability it should probably have at least one